packer/add_IP_property_to_ova.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+set -e
+###################################################
+#Background:
+#   for the purpose of RackHD deployment enviroment without DHCP on admin port (e.x.: eth0),
+#   "OVF Environment" is leveraged to configure VM's IP during OVA/OVF deployment
+#
+#Implementation:
+#   1. define customized fields in OVF template file(XML format)
+#   2. when deploy RackHD OVA , you can specific the eth0 IP (thru OVF Env Parameter) like ```ovftool --prop:adminIP=1.2.3.4 .....```
+#   3. in VM OS's bootup scripts: use VMWare Tool to retrieve IP and set the primary NIC(eth0 for example)
+#
+#  Reference: http://www.v-front.de/2014/01/building-self-configuring-nested-esxi.html
+#
+#Parameter:
+#   the OVA file name to be converted.
+###################################################################
+if  [ ! -n "$1" ];  then
+    echo "[Error] wrong usage of add_IP_property_to_ova.sh. the original ova file name should be given"
+    exit -1
+fi
+OVA=$1
+BASENAME=${OVA%.*}
+OVF=$"${BASENAME}.ovf"
+
+### validate the OVA
+FILE_TYPE=`file ${OVA} | awk -F": " '{print $2}'`
+if [ "${FILE_TYPE}" != "POSIX tar archive" ] ; then
+    echo "[Error]:  ${OVA} is not a valid OVA file"
+    exit -1
+fi
+
+rm -f $"${BASENAME}.mf" # remove checksum file, otherwise, existing mf file will prevent ovftool converting
+echo "convert ${OVA} to OVF file, named ${OVF}"
+ovftool $OVA  $OVF
```

Comment:
I would also add an additional check here in case the ovftool fails
The file can test as a valid tar but still does not necessarily have to be a good OVA for whatever reason
the ovftool tool returns 0 if the conversion was successful, 1 otherwise
if [ $? -eq 1] exit -1